sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -231,11 +231,16 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
         val sizeInBytes = metastoreRelation.statistics.sizeInBytes.toLong
         val fileCatalog = {
           val catalog = new TableFileCatalog(
-            sparkSession, db, table, Some(partitionSchema), sizeInBytes)
+            sparkSession, db, table, Some(partitionSchema), sizeInBytes,
+            fileStatusCacheSize = if (lazyPruningEnabled) {
```

Comment:
shall we inline this logic in `TableFileCatalog`? `TableFileCatalog` can access `SparkSession` so it can know these 2 flags: `filesourcePartitionPruning` and `filesourcePartitionFileCacheSize`